pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -77,14 +77,29 @@ public Response pauseConsumption(
   @POST
   @Path("/tables/{tableName}/resumeConsumption")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation(value = "Resume consumption of a realtime table",
-      notes = "Resume the consumption for a realtime table")
+  @ApiOperation(value = "Resume consumption of a realtime table", notes =
+      "Resume the consumption for a realtime table. ConsumeFrom parameter indicates from which offsets "
+          + "consumption should resume. If consumeFrom parameter is not provided or 'best' option is chosen, "
+          + "consumption continues based on the offsets in segment ZK metadata, and in case the offsets are already "
+          + "gone, the first available offsets are picked to minimize the data loss.")
   public Response resumeConsumption(
-      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName) {
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "earliest | latest | best") @QueryParam("consumeFrom") String consumeFrom) {
     String tableNameWithType = TableNameBuilder.REALTIME.tableNameWithType(tableName);
     validate(tableNameWithType);
+    String offsetCriteria;
+    if ("earliest".equalsIgnoreCase(consumeFrom)) {
+      offsetCriteria = "smallest";
+    } else if ("latest".equalsIgnoreCase(consumeFrom)) {
+      offsetCriteria = "largest";
+    } else if (consumeFrom == null || "best".equalsIgnoreCase(consumeFrom)) {
+      offsetCriteria = null;
+    } else {
```

Comment:
should we have the timestamp and period offset criteria also, now that we support those too?

Reply:
These may not apply to the use cases that we are considering, but for completeness, we can have it (or, we can choose to add later)

Reply:
The change is very simple, but I think we can add it later when there's a request for it.